interface/weno/jsonScript.php

```diff
@@ -0,0 +1,132 @@
+
+$date = date("Y-m-d");
+$pid = $GLOBALS['pid'];
+$uid = $_SESSION['authUserID'];
+
+$i = rand();
+$fillData = filter_input(INPUT_GET, "getJson");
+
+
+$fill = explode(",", $fillData);
+
+$id = $fill[0]; //setting the pharmacy ID for later
+
+array_shift($fill);  //removing the pharmacy from the array
+
+//created a loop in case
+foreach($fill as $data){
+$pInfo = getPatientData($pid);
+
+
+$pfname     = $pInfo['fname'];
+$plname     = $pInfo['lname'];
+$pstreet     = $pInfo['street'];
+$ppostalCode = $pInfo['postal_code'];
+$pcity       = $pInfo['city'];
+$DOB        = $pInfo['DOB'];
+$sex        = $pInfo['sex'];
+
+function fillJsonPatient($pfname,$plname,$pstreet,$ppostalCode,$pcity,$DOB,$sex){
+
+    if($sex == "Male"){ $sex = "M";}
+	if($sex == "Female"){$sex = "F";} //has to be a single letter
+	
+$patient ='[{"patient": {"lname" : "'.$plname.'","fname" : "'.$pfname.'","street" : "'.$pstreet.'","city" : "'.$pcity.'","postal" : "'.$ppostalCode.'","DOB" : "'.$DOB.'","Sex" : "'.$sex.'"}},';
+	return $patient;
+}
+
+$prInfo = new transmitData();
+$proData = $prInfo->getProviderFacility($uid);
+
+    $fname      = $proData[1]['fname'];
+    $lname      = $proData[1]['lname'];
+    $npi        = $proData[1]['npi'];
+    $wenoProvId = $proData[1]['weno_prov_id'];
+    $name       = $proData[1]['name'];
+    $phone      = str_replace("-","",$proData[1]['phone']);
+    $fax        = str_replace("-","",$proData[1]['fax']);
+    $street     = $proData[1]['street'];
+    $city       = $proData[1]['city'];
+    $state      = $proData[1]['state'];
+    $postalCode = $proData[1]['postal_code'];
+    $wenoAccId  = $proData[0][0]['gl_value'];
+    $wenoPass   = $proData[0][1]['gl_value'];
+
+function fillJsonProvider($fname,$lname,$npi,$wenoProvId,$name,$phone,$fax,$street,$city,$state,$postalCode,$wenoAccId,$wenoPass){
+	
+ $provider ='{"provider": {"provlname" : "'.$fname.'","provfname" : "'.$lname.'","provnpi" : '.$npi.',"facilityfax" : '.$fax.',"facilityphone" : '.$phone.',"facilityname" : "'.$name.'","facilitystreet" : "'.$street.'","facilitycity" : "'.$city.'","facilitystate" : "'.$state.'","facilityzip" : '.$postalCode.',"qualifier" : "'.$wenoProvId.'","wenoAccountId" : "'.$wenoAccId.'","wenoAccountPass" : "'.$wenoPass.'","wenoClinicId" : "'.$wenoProvId.'"}},';
+	return $provider; 
+}
+
+$pharmData = $prInfo->findPharmacy($id);
+
+$storename     = $pharmData[0]['name'];
+$storenpi      = $pharmData[0]['npi'];
+$pharmacy      = $pharmData[0]['ncpdp'];
+$pharmacyPhone = $pharmData[1][0]['area_code'].$pharmData[1][0]['prefix'].$pharmData[1][0]['number'];
+$pharmacyFax   = $pharmData[1][1]['area_code'].$pharmData[1][1]['prefix'].$pharmData[1][1]['number'];
+
+function fillJsonPharmacy($storename,$storenpi,$pharmacy,$pharmacyPhone,$pharmacyFax){
+	
+$pharmacy ='{"pharmacy": {"storename" : "'.$storename.'","storenpi" : '.$storenpi.',"pharmacy" : '.$pharmacy.',"pharmacyPhone" : '.$pharmacyPhone.',"pharmacyFax" : '.$pharmacyFax.'}},';
```

Comment:
Wouldn't it be safer to build the php array and then convert it to json via json_encode?(then don't need to worry about escaping values)